lib/SILOptimizer/Transforms/AllocBoxToStack.cpp

```diff
@@ -275,37 +281,72 @@ static bool partialApplyEscapes(SILValue V, bool examineApply) {
 }
 
 static SILInstruction *findUnexpectedBoxUse(SILValue Box,
-                                            bool examinePartialApply,
                                             bool inAppliedFunction,
-                                            SmallVectorImpl<Operand *> &);
+                                            SmallVectorImpl<Operand *> &,
+                                            SmallPtrSetImpl<SILFunction *> &,
+                                            unsigned CurrentRecurDepth);
 
-/// checkPartialApplyBody - Check the body of a partial apply to see
+/// checkLocalApplyBody - Check the body of an apply's callee to see
 /// if the box pointer argument passed to it has uses that would
 /// disqualify it from being promoted to a stack location.  Return
-/// true if this partial apply will not block our promoting the box.
-static bool checkPartialApplyBody(Operand *O) {
+/// true if this apply will not block our promoting the box.
+static bool checkLocalApplyBody(Operand *O,
```

Comment:
In a follow on commit, can you give this a more descriptive name/improve the comments? The big thing here missing from the comment and from the name of the function is that we are /not/ only checking for blocking uses. Specifically, we are also using this to recursively specialize a chain of applies and that since we are visiting recursively, we add the operands (which are in different functions) to the list in DFS order due to the nature of stack recursion.

My hope is that I get that information from the name of this function/the comment. I have a similar issue with findUnexpectedBoxUse. With those names/descriptions it is confusing that you are seeding the algorithm.